src/uct/base/uct_md.c

```diff
@@ -392,6 +441,30 @@ ucs_status_t uct_md_query(uct_md_h md, uct_md_attr_t *md_attr)
     return UCS_OK;
 }
 
+ucs_status_t uct_md_query_v2(uct_md_h md, uct_md_attr_v2_t *md_attr)
+{
+    ucs_status_t status;
+    uct_md_attr_v2_t tmp_md_attr;
+
+    memset(&tmp_md_attr, 0, sizeof(tmp_md_attr));
+
+    /* set all bits */
+    tmp_md_attr.field_mask = -1;
+
+    status = md->ops->query_v2(md, &tmp_md_attr);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    /* Component name + data */
+    memcpy(tmp_md_attr.component_name, md->component->name, sizeof(md_attr->component_name));
+
+    /* based on field mask set by user in md_attr populate only those fields */
```

Comment:
why can't just initialize `md_attr`?

Reply:
Mainly because we want to leave md_attr.field_mask untouched but we also need to set field_mask to -1 to ensure all transports fill all md attributes they have. If we pass md_attr->field_mask to query_v2, then filter function needs to be applied in each of the transport query_v2 functionality.